<?php

require_once '/Users/ivymike/Documents/workspacePHP/NaturalCorner/model/Article.class.php';

/**
 * Test class for Article.
 * Generated by PHPUnit on 2016-02-22 at 22:48:34.
 */
class TestArticle extends PHPUnit_Framework_TestCase
{
    /**
     * @var Article
     */
    protected $article;

    /**
     * Sets up the fixture, for example, opens a network connection.
     * This method is called before a test is executed.
     */
    protected function setUp()
    {
        $this->article = new Article("poire", 150, "Poire bio de Wallonie", true);
    }

    /**
     * Tears down the fixture, for example, closes a network connection.
     * This method is called after a test is executed.
     */
    protected function tearDown()
    {
        $this->article=null;
    }
    /**
     * @covers Article::getId
     * @todo Implement testGetId().
     */
    public function testGetId()
    {
        $this->assertEquals($this->article->getId(), 0, "devrait afficher 0");
         
    }
    /**
     * @covers Article::setId
     * @todo Implement testSetId().
     */
    public function testSetId()
    {
        $this->article->setId(1000);
        $this->assertEquals($this->article->getId(), 1000, "devrait aficher 1000");
         
         
        try{
            $this->article->setId(-1);
        }catch(ArticleException $ue){
            return;
        }
        $this>fail( "aurait dû lancer une exception.");
    
    }
    /**
     * @covers Article::getDenomination
     * @todo Implement testGetDenomination().
     */
    public function testGetDenomination()
    {
    
        $this->assertEquals("poire", $this->article->getDenomination(), "Devrait afficher poire");
    }
    
    /**
     * @covers Article::setDenomination
     * @todo Implement testSetDenomination().
     */
    public function testSetDenomination()
    {
        // tests du setter
        // test de l'exception
    
        try{
            $this->article->setDenomination("NouvellePoire");
        }catch(ArticleException $ue){
    
            $this->fail( "n'aurait pas dû lancer une exception."." ->".$ue);
        }
         
        $this->assertEquals($this->article->getDenomination(), "NouvellePoire", "devrait afficher NouvellePoire");
         
    }
    /**
     * @covers Article::getPrixUnitaire
     * @todo Implement testGetPrixUnitaire().
     */
    public function testGetPrixUnitaire()
    {
    
        $this->assertEquals(150, $this->article->getPrixUnitaire(), "Devrait afficher 150");
    }
    
    /**
     * @covers Article::setPrixUnitaire
     * @todo Implement testSetPrixUnitaire().
     */
    public function testSetPrixUnitaire()
    {
        // tests du setter
        // test de l'exception
    
        try{
            $this->article->setPrixUnitaire(200);
        }catch(ArticleException $ue){
    
            $this->fail( "n'aurait pas dû lancer une exception."." ->".$ue);
        }
         
        $this->assertEquals(200, $this->article->getPrixUnitaire(), "Devrait afficher 200");
         
    }
    /**
     * @covers Article::getCommentaire
     * @todo Implement testGetCommentaire().
     */
    public function testGetCommentaire()
    {
    
        $this->assertEquals("Poire bio de Wallonie", $this->article->getCommentaire(), "Devrait afficher Poire bio de Wallonie");
    }
    
    /**
     * @covers Article::setCommentaire
     * @todo Implement testSetCommentaire().
     */
    public function testSetCommentaire()
    {
        // tests du setter
        // test de l'exception
    
        try{
            $this->article->setCommentaire("Poire bio de Flandre");
        }catch(ArticleException $ue){
    
            $this->fail( "n'aurait pas dû lancer une exception."." ->".$ue);
        }
         
        $this->assertEquals("Poire bio de Flandre", $this->article->getCommentaire(), "Devrait afficher Poire bio de Flandre");
         
    }
    /**
     * @covers Article::isEnVente
     * @todo Implement testIsEnVente().
     */
    public function testIsEnVente()
    {
    
        $this->assertTrue($this->article->isEnVente(), "Devrait afficher true");
    }
    
    /**
     * @covers Article::setEnVente
     * @todo Implement testSetEnVente().
     */
    public function testSetEnVente()
    {
        // tests du setter
        // test de l'exception
    
        try{
            $this->article->setEnVente(false);
        }catch(ArticleException $ue){
    
            $this->fail( "n'aurait pas dû lancer une exception."." ->".$ue);
        }
         
        $this->assertEquals(false, $this->article->isEnVente(), "Devrait afficher false");
         
    }
}
?>




















